dl/src/main/scala/com/intel/analytics/bigdl/dataset/Transformer.scala

```diff
@@ -70,66 +71,164 @@ class Identity[A] extends Transformer[A, A] {
   }
 }
 
+/**
+ * convert sample to MiniBatch, and you can choose whether padding feature or label
+ * to same length in a batch
+ */
```

Comment:
Convert a sequence of Sample to a sequence of MiniBatch, optionally padding all the features (or labels) in the mini-batch to the same length